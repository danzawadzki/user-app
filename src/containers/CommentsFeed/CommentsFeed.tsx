import Accordion from '../Accordion/Accordion';
import CommentsList from '../../components/CommentsList/CommentsList';
import MessageInput from '../../components/MessageInput/MessageInput';
import React from 'react';
import './CommentsFeed.scss';
import { IComment } from '../../components/Comment/Comment';
import WithLoader from '../../components/WithLoader/WithLoader';

export interface ICommentsFeed {
	/** Comments array */
	comments?: IComment[];
	/** Value of comment form */
	commentForm?: string;
	/** Handler to a onChange event. */
	onChange: (e: React.ChangeEvent<HTMLInputElement>) => void;
	/** Handler to a onSubmit event. */
	onSubmit: (e: React.FormEvent<HTMLFormElement>) => void;
}

/**
 * Comments feed component. Contains accordion component and message input to place new posts.
 * @constructor
 */
const CommentsFeed: React.StatelessComponent<ICommentsFeed> = ({
	comments,
	commentForm,
	onChange,
	onSubmit
}) => (
	<>
		<Accordion
			heading={{
				on: `Hide comments (${comments ? comments.length : 0})`,
				off: `Show comments (${comments ? comments.length : 0})`
			}}>
			<CommentsList comments={comments} />
		</Accordion>
		<MessageInput
			id="commentForm"
			value={commentForm}
			placeholder="Add a comment..."
			onChange={onChange}
			onSubmit={onSubmit}
		/>
	</>
);

export default WithLoader(CommentsFeed);
