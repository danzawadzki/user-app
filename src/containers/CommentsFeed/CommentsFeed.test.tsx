import * as React from 'react';
import { mount } from 'enzyme';
import CommentsFeed from './CommentsFeed';

describe('CommentsFeed component: ', () => {
	// Shallow rendered component
	const isLoading = false;
	const commentInput = 'Test';
	const onChange = jest.fn();
	const onSubmit = jest.fn();

	const component = mount(
		<CommentsFeed
			isLoading={isLoading}
			commentForm={commentInput}
			onChange={onChange}
			onSubmit={onSubmit}
		/>
	);

	it('should render match snapshot', () => {
		expect(component).toMatchSnapshot();
	});

	it('should fire onSubmit func', () => {
		expect(onSubmit).toHaveBeenCalledTimes(0);
		expect(component.find('form.MessageInput')).toBeTruthy();
		expect(component.find('form.MessageInput').simulate('submit'));
		expect(onSubmit).toHaveBeenCalledTimes(1);
	});
	it('should fire onSubmit func', () => {
		expect(onChange).toHaveBeenCalledTimes(0);
		expect(component.find('input')).toBeTruthy();
		expect(component.find('input').simulate('change'));
		expect(onChange).toHaveBeenCalledTimes(1);
	});
});
