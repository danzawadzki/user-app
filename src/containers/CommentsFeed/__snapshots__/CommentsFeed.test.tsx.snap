// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommentsFeed component:  should render match snapshot 1`] = `
<WithLoading
  commentForm="Test"
  isLoading={false}
  onChange={[MockFunction]}
  onSubmit={[MockFunction]}
>
  <CommentsFeed
    commentForm="Test"
    onChange={[MockFunction]}
    onSubmit={[MockFunction]}
  >
    <Accordion
      heading={
        Object {
          "off": "Show comments (0)",
          "on": "Hide comments (0)",
        }
      }
    >
      <div
        className="Accordion"
      >
        <a
          className="Accordion__heading"
          onClick={[Function]}
        >
          Hide comments (0)
        </a>
        <div
          className="Accordion__body Accordion__body--visible"
        >
          <CommentsList>
            <p
              className="CommentsList__message"
            >
              No comments...
            </p>
          </CommentsList>
        </div>
      </div>
    </Accordion>
    <MessageInput
      id="commentForm"
      onChange={[MockFunction]}
      onSubmit={[MockFunction]}
      placeholder="Add a comment..."
      value="Test"
    >
      <form
        className="MessageInput"
        onSubmit={[MockFunction]}
      >
        <input
          className="MessageInput__input"
          id="commentForm"
          onChange={[MockFunction]}
          placeholder="Add a comment..."
          value="Test"
        />
      </form>
    </MessageInput>
  </CommentsFeed>
</WithLoading>
`;
